crates/wasi-nn/src/lib.rs

```diff
@@ -42,3 +61,141 @@ impl std::ops::DerefMut for ExecutionContext {
+        Ok(())
+    }
+
+    /// Return `Ok` if we find a working OpenVINO installation.
+    fn check_openvino_is_installed() -> Result<()> {
+        match std::panic::catch_unwind(|| {
+            println!("> found openvino version: {}", openvino::version())
+        }) {
+            Ok(_) => Ok(()),
+            Err(e) => Err(anyhow!("unable to find an OpenVINO installation: {:?}", e)),
+        }
+    }
+
+    /// Return `Ok` if we find the cached MobileNet test artifacts; this will
+    /// download the artifacts if necessary.
+    fn check_openvino_artifacts_are_available() -> Result<()> {
+        const BASE_URL: &str = "https://github.com/intel/openvino-rs/raw/main/crates/openvino/tests/fixtures/mobilenet";
+        let artifacts_dir = artifacts_dir();
+        if !artifacts_dir.is_dir() {
+            fs::create_dir(&artifacts_dir)?;
+        }
+        for (from, to) in [
+            ("mobilenet.bin", "model.bin"),
+            ("mobilenet.xml", "model.xml"),
+            ("tensor-1x224x224x3-f32.bgr", "tensor.bgr"),
+        ] {
+            let remote_url = [BASE_URL, from].join("/");
+            let local_path = artifacts_dir.join(to);
+            if !local_path.is_file() {
+                download(&remote_url, &local_path)
+                    .with_context(|| "unable to retrieve test artifact")?;
+            } else {
+                println!("> using cached artifact: {}", local_path.display())
+            }
+        }
+        Ok(())
+    }
+
+    /// Retrieve the bytes at the `from` URL and place them in the `to` file.
+    fn download(from: &str, to: &Path) -> anyhow::Result<()> {
+        println!("> downloading:\n  {} ->\n  {}", from, to.display());
+        let mut file = File::create(to)?;
+        let _ = reqwest::blocking::get(from)?.copy_to(&mut file)?;
+        Ok(())
+    }
+
+    /// Build the given crate as `wasm32-wasi` and return the path to the built
+    /// module.
+    pub fn cargo_build(crate_dir: impl AsRef<Path>) -> PathBuf {
+        let crate_dir = crate_dir.as_ref();
+        let crate_name = crate_dir.file_name().unwrap().to_str().unwrap();
+        let cargo_toml = crate_dir.join("Cargo.toml");
+        let wasm = crate_dir.join(format!("target/wasm32-wasi/release/{}.wasm", crate_name));
+        let result = Command::new("cargo")
+            .arg("build")
+            .arg("--release")
+            .arg("--target=wasm32-wasi")
+            .arg("--manifest-path")
+            .arg(cargo_toml)
+            .output()
+            .unwrap();
+        if !wasm.is_file() {
+            panic!("no file found at: {}", wasm.display());
+        }
+        if !result.status.success() {
+            panic!(
+                "cargo build failed: {}\n{}",
+                result.status,
+                String::from_utf8_lossy(&result.stderr)
+            );
+        }
+        wasm
+    }
+}
+
+#[cfg(all(test, not(feature = "test-check")))]
+compile_error!(
```

Comment:
Is the `test-check` feature here primarily for the `reqwest` dependency? If that goes away could this part go away to and unconditionally use `cfg(test)` for the `test_check` stuff?

Reply:
Well, kind of. I need to be able to use the `test_check` module from various places: unit tests, integration tests, Wasmtime CLI tests... When I was trying to make this work I observed that I could `#[cfg(test)]` the `test_check` module and still use it for unit tests but not for the other cases. I actually need this logic available (somehow) in the public interface of the crate. Or perhaps not "_the_ crate" but rather "_a_ crate": I could move this logic to a separate crate like `wasmtime-wasi-nn-test` and create a `dev-dependency` on that?

Reply:
Ah ok yeah makes sense, and yes if it's required outside this crate then `#[cfg(test)]` won't work. That being said though since there's no dependencies behind this implementation any more with `reqwest` replaced I think it'd be fine to unconditionally include this functionality in the crate and other embedders would largely just ignore it.